core/src/main/scala/com/acervera/osm4scala/DenseNodesIterator.scala

```diff
@@ -26,78 +26,115 @@
 package com.acervera.osm4scala
 
 import com.acervera.osm4scala.model.NodeEntity
-import org.openstreetmap.osmosis.osmbinary.osmformat.{DenseNodes, StringTable}
-
-object DenseNodesIterator {
-
-  def apply(osmosisStringTable: StringTable, osmosisDenseNode: DenseNodes): DenseNodesIterator =
-    new DenseNodesIterator(osmosisStringTable, osmosisDenseNode)
-
-}
+import com.acervera.osm4scala.utilities.DecompressUtils
+import com.acervera.osm4scala.utilities.DecompressUtils.{decompressChangeset, decompressCoord, decompressTimestamp, decompressUid, decompressUserSid, iteratorCheck}
+import org.openstreetmap.osmosis.osmbinary.osmformat.{DenseInfo, DenseNodes, StringTable}
 
 /**
   * Iterator over a DenseDataNode block.
-  * By default, lanOffset, longOffset and graularity is 0, 0 and 100 because I did not found pbf files with other values.
   *
   * @param osmosisStringTable
   * @param osmosisDenseNode
   * @param latOffset
   * @param lonOffset
   * @param granularity
+  * @param dateGranularity
   */
-class DenseNodesIterator(osmosisStringTable: StringTable,
+case class DenseNodesIterator(osmosisStringTable: StringTable,
```

Comment:
Case classes represent immutable entities. An iterator is never immutable. Every `next` the class will mutate.

Reply:
I think the densenode iterator here only iterate within the class and it's not a iterator indeed. Using the case class would be better for serialization. I think we can reconsider the class name here.

Reply:
Yes, case classes are not the problem. The problem is that this is not the case. When you create a case class, Scala creates under the cover a lot of stuff always with the idea of immutability. This is not the case at all. If you do this, if one user try to use `.copy`, `pattern matching` or any defaul `apply` or `unapply`, `equal`, `hash`, etc... is going to have problems because nothing of that will work.
In fact, because this, this is a bug. An Iterator is not a case class at all.

But the more important point, this modification is out of scope for this ticket.

Reply:
The first argument make sense. I can rollback. But it's within the scope.